exercises/queen-attack/src/example/java/QueenAttackCalculator.java

```diff
@@ -0,0 +1,86 @@
+import java.util.Arrays;
+
+public final class QueenAttackCalculator {
+
+    private final int[] queen1Coordinates;
+
+    private final int[] queen2Coordinates;
+
+    public QueenAttackCalculator(final int[] queen1Coordinates, final int[] queen2Coordinates)
```

Comment:
R.E. a `Coordinate` type - I agree that primitive types are a little ugly here, but I avoided introducing a second separate type that requires implementation as I did not find another example which follows that pattern. (This is probably why other exercises used e.g. Guava to provide a nicer parameter type, though I agree with the removal of that third party dependency moving forward). It shouldn't be hard to adapt the problem, however, so I'll make the change to use a separate `Coordinate` type and we can see how that feels!

R.E. renaming to white/black - yep, great call, will make the change!

R.E. notation change - there's a note about using chess notation in the canonical data set that mentions it, but IMO it's not really clear how the note should be interpreted:

> Some languages implement tests beyond this set, such as checking for two pieces being placed on the same position, representing the board graphically, or using standard chess notation. Those tests can be offered as extra credit.

Is there build-in support in Exercism for optional/extra credit tests? If not, I'm guessing this means we could expose two methods that accept differently-formatted `Coordinate` types and require implementations that satisfy both to consider the problem complete. That feels like relatively uninteresting duplication, however.